WordPress/Classes/ViewRelated/NUX/SignupDomainSuggestionViewController.swift

```diff
@@ -0,0 +1,244 @@
+            }
+        case Sections.searchSuggestions.rawValue:
+            if isSearching {
+                return 1
+            } else {
+                return searchSuggestions.count
+            }
+        case Sections.siteTitleSuggestions.rawValue:
+            return siteTitleSuggestions.count
+        default:
+            return 0
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell: UITableViewCell
+        switch indexPath.section {
+        case Sections.titleAndDescription.rawValue:
+            cell = titleAndDescriptionCell()
+        case Sections.searchField.rawValue:
+            cell = searchFieldCell()
+        case Sections.searchSuggestions.rawValue:
+            if isSearching {
+                cell = activityCell()
+            } else {
+                cell = searchButtonCell(index: indexPath.row)
+            }
+        case Sections.siteTitleSuggestions.rawValue:
+            fallthrough
+        default:
+            cell = buttonCell(index: indexPath.row)
+        }
+        return cell
+    }
+
+    override func tableView(_ tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
+        guard section == Sections.siteTitleSuggestions.rawValue || section == Sections.searchField.rawValue else {
+            return nil
+        }
+        let footer = UIView()
+        footer.backgroundColor = WPStyleGuide.greyLighten20()
+        return footer
+    }
+
+    override func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        guard section == Sections.siteTitleSuggestions.rawValue || section == Sections.searchField.rawValue else {
+            return 0.0
+        }
+        return 0.5
+    }
+
+    // MARK: table view cells
+
+    private func activityCell() -> SiteCreationDomainsActivityTableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: SiteCreationDomainsActivityTableViewCell.cellIdentifier) as? SiteCreationDomainsActivityTableViewCell else {
+            let newCell = SiteCreationDomainsActivityTableViewCell(style: .default, reuseIdentifier: SiteCreationDomainsActivityTableViewCell.cellIdentifier)
+            newCell.activitySpinner?.startAnimating()
+            return newCell
+        }
+        cell.activitySpinner?.startAnimating()
+        return cell
+    }
+
+    private func titleAndDescriptionCell() -> UITableViewCell {
+        let title = NSLocalizedString("Step 4 of 4", comment: "Title for last step in the site creation process.").localizedUppercase
+        let description = NSLocalizedString("Pick an available \"yourname.wordpress.com\" address to let people find you on the web.", comment: "Description of how to pick a domain name during the site creation process")
+        let cell = LoginSocialErrorCell(title: title, description: description)
+        cell.selectionStyle = .none
+        return cell
```

Comment:
The text formatting of this cell doesn't match the prior steps. I assume this is left to the future task of merging those steps cells and LoginSocialErrorCell?